network/p2p/p2pbuilder/libp2pNodeBuilder.go

```diff
@@ -391,20 +365,31 @@ func (builder *LibP2PNodeBuilder) Build() (p2p.LibP2PNode, error) {
 
 	cm := component.NewComponentManagerBuilder().
 		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			// routing system is created here, because it needs to be created during the node startup.
 			rsys, err := builder.buildRouting(ctx, h)
 			if err != nil {
-				ctx.Throw(fmt.Errorf("could not create routing system: %w", err))
+				ctx.Throw(fmt.Errorf("could not create rsys system: %w", err))
 			}
 			node.SetRouting(rsys)
+			builder.gossipSubBuilder.SetRoutingSystem(rsys)
 
-			gossipSub, err := builder.buildGossipSub(ctx, rsys, h)
+			// gossipsub is created here, because it needs to be created during the node startup.
+			gossipSub, scoreTracer, err := builder.gossipSubBuilder.Build(ctx)
 			if err != nil {
 				ctx.Throw(fmt.Errorf("could not create gossipsub: %w", err))
 			}
+			if scoreTracer != nil {
+				node.SetPeerScoreExposer(scoreTracer)
```

Comment:
Should this method wait for `scoreTracer.Done()` before returning?

Reply:
https://github.com/onflow/flow-go/pull/3986/commits/0600d5c82f5a8ddeb311911a316826b08b0ac514